src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+        } else {
+            Err(anyhow!(
+                "Invalid continuation: {}",
+                cont_key.fmt_to_string(store, &repl.state.borrow())
+            ))
+        }
+    }
+
+    const PROVEPROTOCOL: MetaCmd<F> = MetaCmd {
+        name: "prove-protocol",
+        summary: "Creates a proof for a protocol",
+        format: "!(prove-protocol <protocol> <string> args...)",
+        description: &[
+            "The proof is created only if the protocol can be satisfied by the",
+            "  provided arguments.",
+            "The second (string) argument for this meta command is the path to",
+            "  the file where the protocol proof will be saved.",
+        ],
+        example: &[
+            "(commit '(13 . 17))",
+            "!(prove-protocol my-protocol",
+            "  \"protocol-proof\"",
+            "  0x09910d31a7568d66855bcc83fccc4826063dfdf93fe5e1f736c83ec892ed139e",
+            "  '(13 . 17))",
+        ],
+        run: |repl, args| {
+            let (ptcl, rest) = repl.store.car_cdr(args)?;
+            let (path, args) = repl.store.car_cdr(&rest)?;
+
+            let path = get_path(repl, &path)?;
+
+            let (vars, body, proto_rc) = Self::get_vars_body_and_rc(repl, ptcl)?;
+
+            if proto_rc != repl.rc {
+                bail!(
+                    "Protocol uses rc={proto_rc} but the REPL was initialized with rc={}",
+                    repl.rc
+                )
+            }
+
+            let args_vec = Self::get_args_vec(repl, &args, vars.len())?;
+
+            let mut args_vec_evaled = Vec::with_capacity(args_vec.len());
+            for a in args_vec {
+                let (io, ..) = repl.eval_expr(a)?;
+                args_vec_evaled.push(io[0]);
+            }
+
+            // shadowing original `args` with evaluated ones
+            let args = repl.store.list(args_vec_evaled.clone());
+
+            let (cek_io, post_verify) =
+                Self::get_cek_io_and_post_verify_function(repl, vars, args_vec_evaled, body)?;
+
+            Self::post_verify_check(repl, &post_verify)?;
+
+            let (frames, iterations) = evaluate_with_env_and_cont::<F, Coproc<F>>(
+                None,
+                cek_io[0],
+                cek_io[1],
+                Self::get_cont_ptr(repl, &cek_io[2])?,
+                &repl.store,
+                repl.limit,
+            )?;
+
+            {
+                // making sure the output matches expectation before proving
+                let res = &frames.last().expect("frames can't be empty").output;
+                if cek_io[3] != res[0]
+                    || cek_io[4] != res[1]
+                    || Self::get_cont_ptr(repl, &cek_io[5])? != res[2]
+                {
+                    bail!("Mismatch between expected output and computed output")
+                }
+            }
+
+            repl.evaluation = Some(Evaluation { frames, iterations });
+            let proof_key = repl.prove_last_frames()?;
```

Comment:
I guess it's fine since this is just a convenience, but I don't really think this should be relying on REPL state in this way.